lib/webhooks/validate.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  abstractConvertRequest,
+  getHeader,
+  Headers,
+  NormalizedRequest,
+} from '../../runtime/http';
+import {createSHA256HMAC} from '../../runtime/crypto';
+import {HashFormat} from '../../runtime/crypto/types';
+import {ShopifyHeader} from '../types';
+import {ConfigInterface} from '../base-types';
+import {safeCompare} from '../auth/oauth/safe-compare';
+import {logger} from '../logger';
+
+import {
+  WebhookFields,
+  WebhookValidateParams,
+  WebhookValidation,
+  WebhookValidationErrorReason,
+} from './types';
+import {topicForStorage} from './registry';
+
+const HANDLER_PROPERTIES: {
+  [property in keyof WebhookFields]: ShopifyHeader;
+} = {
+  apiVersion: ShopifyHeader.ApiVersion,
+  domain: ShopifyHeader.Domain,
+  hmac: ShopifyHeader.Hmac,
+  topic: ShopifyHeader.Topic,
+  webhookId: ShopifyHeader.WebhookId,
+};
+
+export function validate(config: ConfigInterface) {
```

Comment:
Should we start calling these _factories_ like we do in the _other_ project 😉 , e.g., 
```suggestion
export function validateFactory(config: ConfigInterface) {
```